### PR TITLE
Reference ToC refactor

### PR DESCRIPTION
**Pull request by chenopis**
Reorganize contents of Reference section and its ToC.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3427)

### CODE REVIEW

File: _data/reference.yml [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
+    - title: API
+      path: /docs/api-reference/v1.6/
+    - title: Resource Types
+      path: /docs/resources-reference/v1.6/
   - title: Version 1.5
-    path: /docs/resources-reference/v1.5/
+    section:
+    - title: API
+      path: /docs/api-reference/v1.5/
+    - title: Resource Types
```
Comment (ahmetb) [inline _data/reference.yml]: can we get rid of /resources-reference ones?

File: _data/reference.yml [hunk 2]
```diff
@@ -1,41 +1,62 @@
...
-- docs/reference.md
+- docs/reference/index.md
 
-- title: "Kubernetes Resource Types"
+- title: "API Reference"
   section:
   - title: Version 1.6
-    path: /docs/resources-reference/v1.6/
+    section:
+    - title: API
```
Comment (ahmetb) [inline _data/reference.yml]: I think it should go like this:

- API Reference
   - v1.6 API
   - v1.5 API

File: _data/reference.yml [hunk 3]
```diff
@@ -1,41 +1,62 @@
...
+    - title: Resource Types
+      path: /docs/resources-reference/v1.6/
   - title: Version 1.5
-    path: /docs/resources-reference/v1.5/
+    section:
+    - title: API
+      path: /docs/api-reference/v1.5/
+    - title: Resource Types
+      path: /docs/resources-reference/v1.5/
+  - title: Federation
```
Comment (ahmetb) [inline _data/reference.yml]: this section doesn't fit here. There's a single `Federation` title under `API` section, where are the others (apps, extensions, etc)? This is confusing. Maybe move it under the top level and call it `Federation API`.

File: _data/reference.yml [hunk 4]
```diff
@@ -1,41 +1,62 @@
...
     - docs/admin/service-accounts-admin.md
-  - docs/api-reference/labels-annotations-taints.md
-  - docs/api-reference/extensions/v1beta1/definitions.html
+  - title: Authorization Plugins
+    section:
+    - docs/admin/authorization/index.md
+    - docs/admin/authorization/rbac.md
 
-- title: kubectl CLI
+- title: Working with Kubernetes
```
Comment (ahmetb) [inline _data/reference.yml]: this should be `Working with kubectl CLI`

File: _data/reference.yml [hunk 5]
```diff
@@ -49,24 +70,20 @@ toc:
...
-  - docs/admin/kube-apiserver.md
-  - docs/admin/kube-controller-manager.md
-  - docs/admin/kube-proxy.md
-  - docs/admin/kube-scheduler.md
-  - docs/admin/kubeadm.md
-  - docs/admin/federation-apiserver.md
-  - docs/admin/federation-controller-manager.md
-  - docs/federation/api-reference/index.md
-  - title: kubelet
+  - title: kubectl CLI Reference
```
Comment (ahmetb) [inline _data/reference.yml]: currently this does:

- Kubernetes components
  - kubectl CLI Reference
     - apiserver
     - kube-proxy
     - ...anything but kubectl

doesn't make sense. Can we just create a title `Command-line tools` and put things in here. We don't talk about kubectl here.

File: _data/reference.yml [hunk 6]
```diff
@@ -1,73 +1,88 @@
...
     - docs/admin/service-accounts-admin.md
-  - docs/api-reference/labels-annotations-taints.md
-  - docs/api-reference/extensions/v1beta1/definitions.html
+  - title: Authorization Plugins
+    section:
+    - docs/admin/authorization/index.md
+    - docs/admin/authorization/rbac.md
 
-- title: kubectl CLI
+- title: Working with Kubernetes using kubectl
```
Comment (ahmetb) [inline _data/reference.yml]: I am still reluctant to call it `Working with Kubernetes using kubectl`.

If I'm reading reference docs I already know one uses kubectl to "work with kubernetes". So this title sounds a bit obsolete to me.

It also feels like tutorials & tasks articles such as the following should be listed here if the title is `Working with Kubernetes using kubectl`:

-  Managing Kubernetes Objects Using Imperative Commands
- Installing and Setting Up kubectl
- Listing All Container Images Running in a Cluster

but clearly they don't belong here.

**Comment by chenopis** [comment]
So this begs the question of whether we should present it as several APIs or one API with several groups. If the latter, then I think Federation needs to be under API Reference as well. I agree it's too nested at the moment, but is there a way we can refactor the Federation docs?

File: _data/reference.yml (hunk 6 above)
Comment (chenopis) [inline _data/reference.yml]: Should we just call the section kubectl? Most people will probably know what kubectl is, so they need to be able to find it quickly.

**Comment by ahmetb** [comment]
other APIs are under the main API reference doc, but it appears like Federation API is a horizontal effort, so it spans multiple API categories. Let’s move federation API to the top level and review how it looks like.

**Comment by chenopis** [comment]
@ahmetb Ok, I incorporated your feedback. PTAL

File: _data/reference.yml [hunk 7]
```diff
@@ -1,53 +1,60 @@
...
-  section:
-  - title: Overview
-    path: /docs/concepts/overview/kubernetes-api/
-  - title: Version 1.6
+  - title: v1.6
     path: /docs/api-reference/v1.6/
-  - title: Version 1.5
+  - title: v1.5
     path: /docs/api-reference/v1.5/
+  - docs/reference/extensions/v1beta1/definitions.html
```
Comment (ahmetb) [inline _data/reference.yml]: I don't think this page is necessary. Also it's not a release version like v1.6.

The huge v1.6 API reference page covers everything this /v1beta1/ page has. cc:@pwittrock to confirm

Comment (chenopis) [inline _data/reference.yml]: I'm happy to get rid of that. =)

**Comment by jbeda** [comment]
The CLI references are a bit confusing.  kubectl is really it is what I would call a command line tool.  It is the type of thing that folks will use consistently. kubeadm and kubefed would apply here also.

The other things and that list seem like a random grab bag.  The only time that folks will be looking at this is when setting up or reconfiguring a cluster.  They aren't a day to day thing and they are targeted at a completely different audience.

I'd move those set into a separate section on this page or strike them altogether.  Furthermore, in the ToC, I'd name the section that has kube-apiserver et.al. to be named "Server Binary Reference" or something.  It really isn't a CLI as much as a config reference.

Finally, there are sub-docs for kubelet that are really out of place here.  These are more reference vs. just command line flags.  They also have to do with auth.  Not sure how to reconcile this doc (https://deploy-preview-3427--kubernetes-io-master-staging.netlify.com/docs/admin/kubelet-tls-bootstrapping/) with this doc (https://deploy-preview-3427--kubernetes-io-master-staging.netlify.com/docs/admin/authentication/).

File: docs/reference/index.md [hunk 8]
```diff
@@ -4,7 +4,10 @@ assignees:
 - chenopis
 ---
 
-In the reference section, you can find reference documentation for Kubernetes APIs, CLIs, and tools, as well as our glossary and design docs. 
+The Reference section contains detailed documentation on the Kubernetes [API](/docs/api-reference/{{page.version}}/),
+[kubectl](/docs/user-guide/kubectl-overview/) command-line interface (CLI),
+[kubeadm](/docs/admin/kubeadm/) setup tool,
+configuration reference for various Kubernetes components, design docs, and issues and security.
```
Comment (ahmetb) [inline docs/reference/index.md]: > design docs, and issues and security.

this is not necessary IMHO.

**Comment by ahmetb** [comment]
@chenopis 
I'm not in favor of using `API Overview` page as the landing page for the Reference section. There's much more to it. You can move `API Overview` from Concepts to Reference, but making it the landing page doesn't make sense to me, as not everything in /docs/reference is about the API. Consider rolling back 6b98d3f.

**Comment by ahmetb** [comment]
I'm okay with commits in this patch including 05a9119, but not the 6b98d3f as it doesn't reflect what we agreed on. It's getting a bit harder to review as this PR started as reasonably small, then got bigger and bigger. Let's limit the scope of it and postpone  major changes that aren't discussed to another PR. I suggest reverting to 05a9119 and merging.

**Comment by chenopis** [comment]
@ahmetb I was just trying out some changes based on some feedback. I agree though, so will revert to 05a9119.

**Comment by chenopis** [comment]
@ahmetb Ok, this should be the final state. PTAL and give me the final LGTM so that I can squash and merge this. Thx

**Pull request merged**